### ISSUE TITLE
Support for Juniper SRX Clusters

### ISSUE THREAD

**Issue by domel138** [digitalocean/netbox#1959]
### Issue type
[x] Feature request 
[ ] Bug report      
[ ] Documentation   

### Environment
* Python version: 3.5.5
* NetBox version: 2.3.1

### Description
Im wondering if you were considering adding support for Juniper SRX Firewall Clusters. Basically they relay on reth (redundant Ethernet interfaces) - ideally i would see this implemented in similar way that Virtual Chassis is (single control-plane) but there is couple of caveat's like reth interfaces, management ip etc. 
I would like to have for this functinality to have:
- in-band management via reth interface sharing same ip-address across two nodes
- out-of-band management via fxp0 with separate ip-addresses (it is possible now i believe)
- sharing any ip-address across reth interface with virtual mac-address

Basically i think it should be simillar to Virtual Chassis with different roles - instead of master/members it should say node0/node1 with respective roles 

Use case would be to add SRX clusters into Netbox - have no idea on what changes on postgres end this will require but maybe someone will help with it

**Comment by PeepOks** [comment]
In some way it's similar to https://github.com/digitalocean/netbox/issues/1950

**Comment by jeremystretch** [comment]
Any representation of SRX clusters will need to be done within the virtual chassis model. We can look at tweaking it somewhat if needed, but we definitely won't be adding any new models to support this.

**Comment by domel138** [comment]
Hi Jeremy,

Sorry im completely new at github society - however i can take a look at VC
implementation and propose some enhancements to support SRX platform - is
it a good way to comment it via github request or shall i create new thread
for this?

Pozdrawiam, Dominik Siecinski.

2018-03-09 20:47 GMT+01:00 Jeremy Stretch <notifications@github.com>:

**Comment by DanSheps** [comment]
My personal opinion on the virtual chassis setup right now, is it seems to work best for something like a switch stack, where there is 1 management and control plane (and a semi-shared data plane).

I think the current VC model fails for things like vPC, VSS, SRX clusters and ASA clusters, where there are sometimes independent interfaces on all devices but also can be shared interfaces (or HSRP/VRRP/GLBP/etc interfaces tied to both devices).

I am not sure the best way to correct this "problem" though, as a tweak the other way would not be suitable for stacked switches.  Perhaps a VC model "type" that changes the behavior of the VC based on the type (for example, vPC/VSS would not show shared interfaces, might add things like vPC domain and such).  This seems like a big job though and I honestly think it might be easier to try a parent/child device relationship where the parent is a psuedo device and the child is the individual devices.  Not sure how that works out either though.

That said, with the virtual chassis, I do wish it was represented as a device itself, instead of the master device showing all interfaces, as this makes it difficult for quickly adding connections from another device to the master unit (slave units you can individually pick and set the interface from the interface list without seeing all interfaces).

Just my thoughts, I am sure I have more rattling around in there as well...

**Comment by luqasz** [comment]
> That said, with the virtual chassis, I do wish it was represented as a device itself, instead of the master device showing all interfaces, as this makes it difficult for quickly adding connections from another device to the master unit (slave units you can individually pick and set the interface from the interface list without seeing all interfaces).

Same here. I have a stack of juniper EX4300 and I'd like to add interfaces to virtual chassis and not individual nodes.

**Comment by jeremystretch** [comment]
> That said, with the virtual chassis, I do wish it was represented as a device itself, instead of the master device showing all interfaces

This is the way NetBox represents a virtual chassis. If you need to model virtual chassis differently, NetBox may not be the right tool for you.

**Comment by jeremystretch** [comment]
Closing this out as no one has proposed a specific model for representing an SRX cluster or similar arrangement.

**Comment by luqasz** [comment]
> Closing this out as no one has proposed a specific model for representing an SRX cluster or similar arrangement.

What about assigning ips, interfaces etc to a virtual chassis ?
